astroquery/nasa_exoplanet_archive/nasa_exoplanet_archive.py

```diff
@@ -98,28 +103,70 @@ def get_confirmed_planets_table(self, cache=True, show_progress=True,
                         exoplanets_table[col].unit = u.Unit(self.param_units[col])
 
             self._table = QTable(exoplanets_table)
+            self._cache_planet = planet_name
 
         return self._table
 
-    def query_planet(self, planet_name, **kwargs):
+    def query_star(self, host_name, cache=True, show_progress=True,
+                        table_path=None, all_columns=False):
         """
-        Get table of exoplanet properties.
+        Download (and optionally cache) the NASA Exoplanet
+        table for a certain planet.
 
         Parameters
         ----------
-        planet_name : str
-            Name of planet
-        kwargs : dict (optional)
-            Extra keyword arguments passed to ``get_confirmed_planets_table``.
+        host_name : str
+            Name of stellar system
+        cache : bool (optional)
+            Cache exoplanet table to local astropy cache? Default is `True`.
+        show_progress : bool (optional)
+            Show progress of exoplanet table download (if no cached copy is
+            available). Default is `True`.
+        table_path : str (optional)
+            Path to a local table file. Default `None` will trigger a
+            download of the table from the internet.
+        all_columns : bool (optional)
+            Return all available columns. The default returns only the
+            columns in the default category at the link above.
 
         Returns
         -------
         table : `~astropy.table.QTable`
             Table of one exoplanet's properties.
         """
 
-        exoplanet_table = self.get_confirmed_planets_table(**kwargs)
-        return exoplanet_table.loc[planet_name.strip().lower().replace(' ', '')]
+        host_name = '\'' + host_name + '\''
+        EXOPLANET_URL_PLANET = EXOPLANETS_CSV_URL + '&where=pl_hostname=' + quote(host_name)
+
+        if self._table is not None and self._cache_star is host_name and cache:
+            pass
```

Comment:
this block would be a little clearer if you had `return self._table` here instead of `pass`ing.  Even though it repeats a line of code, it makes it clearer what this block of text is doing.